drivers/volume/portworx/portworx.go

```diff
@@ -2438,6 +2438,75 @@ func (p *portworx) DeletePair(pair *storkapi.ClusterPair) error {
 	return nil
 }
 
+func (p *portworx) Failover(action *storkapi.Action) error {
+	namespace := action.Namespace
+	logrus.Debugf("Failover: namespace %s", namespace)
+	volDriver, err := p.getAdminVolDriver() // TODO(dgoel): admin or user vol driver?
+	if err != nil {
+		return err
+	}
+
+	// TODO(dgoel): when should we return error
+	// and should we just log an error and continue
+
+	// TODO(dgoel): show relevant events in the Action CR
+	// return list of errors (as above) - controller can use to log events
+	// for ex: a volume promote failed
+
+	// can we directly fetch PersistentVolumes here?
+	pvcList, err := core.Instance().GetPersistentVolumeClaims(namespace, nil)
+	if err != nil {
+		return fmt.Errorf("error getting list of volumes to migrate: %v", err)
+	}
+	for _, pvc := range pvcList.Items {
+		if !p.OwnsPVC(core.Instance(), &pvc) {
+			continue
+		}
+		if resourcecollector.SkipResource(pvc.Annotations) { // TODO(dgoel): do we need this?
```

Comment:
Yes we should keep this, so that if a user wants to not migrate a PVC object within a given namespace.